src/listener/tcp_listener.rs

```diff
@@ -0,0 +1,80 @@
+use super::is_transient_error;
+
+use crate::listener::Listener;
+use crate::utils::BoxFuture;
+use crate::{log, Server};
+
+use async_std::net::{self, SocketAddr, TcpStream};
+use async_std::prelude::*;
+use async_std::{io, task};
+
+#[derive(Debug)]
+pub enum TcpListener {
+    FromListener(net::TcpListener),
```

Comment:
This is an enum in order to avoid binding to a string spec until .listen is called.
```rust
let mut multi = MultiListener::new();
multi.add("http://localhost:3000");
//at this point, 3000 should not be bound
app.listen(multi) // but now it is
```